gateway/api/use_cases/get_jobs.py

```diff
@@ -0,0 +1,54 @@
+"""This module contains the usecase get_jos"""
+import logging
+from typing import List, Optional
+from datetime import datetime
+from api.models import Job
+from api.repositories.jobs import JobsRepository, JobFilters
+from api.views.enums.type_filter import TypeFilter
+
+logger = logging.getLogger("gateway.use_cases.get_jobs")
+
+
+class GetJobsUseCase:
+    """Use case for retrieving user jobs with optional filtering and pagination."""
+
+    jobs_repository: JobsRepository = JobsRepository()
+
+    def __init__(  # pylint:  disable=too-many-positional-arguments
+        self,
+        user,
+        limit: Optional[int],
+        offset: int = 0,
+        type_filter: Optional[TypeFilter] = None,
+        status: Optional[str] = None,
+        created_after: Optional[datetime] = None,
+        function_name: Optional[str] = None,
+    ):
+        self.user = user
+        self.limit = limit
+        self.offset = offset
+        self.type_filter = type_filter
+        self.status = status
+        self.created_after = created_after
+        self.function_name = function_name
```

Comment:
I have two comments here:

 - To remove the "too many positional arguments" maybe you can use the `JobRepository` >`JobFilter` `dataclass`. I think it will also be more readable.
 - I was migrating the `files` endpoint and I put the args in the execute instead of the init. I think it is good to have consensus on this. In my opinion, most of this parameters are not a "permanent" state, are more related with the action. Also it simplifies the code, we don't have to set object variables, just use them.

What do you think?